src/chartdata/ChartData.js

```diff
@@ -0,0 +1,879 @@
+        // are in both frames
+        const frameA: List<ChartRow> = frames
+            .get(indexA)
+            .groupBy(ii => ii.get(primaryColumn));
+
+        const frameB: List<ChartRow> = frames
+            .get(indexB)
+            .groupBy(ii => ii.get(primaryColumn));
+
+        const getRowFromFrame = (frame: List<ChartRow>, primaryValue: ChartScalar): ?ChartRow => {
+            const rowList: ?List<ChartRow> = frame.get(primaryValue);
+            if(!rowList || rowList.size == 0) {
+                return null;
+            }
+            const first: ChartRow = rowList.first();
+            // when uniquely identifying points, the data may accidentally contain multiple matches,
+            // so warn if this happens
+            if(rowList.size > 1) {
+                console.warn(`ChartData: Two data points found where ${frameColumn}=${first.get(frameColumn)} and ${primaryColumn}=${String(primaryValue)}, using first data point.`);
+            }
+            return first;
+        };
+
+        // for all primaryValues, find matching data points in both frames
+        const interpolatedFrame: List<ChartRow> = allPrimaryValues
+            .map((primaryValue: ChartScalar): ?ChartRow => {
+                const rowA: ?ChartRow = getRowFromFrame(frameA, primaryValue);
+                const rowB: ?ChartRow = getRowFromFrame(frameB, primaryValue);
+
+                // one of the data points doesn't exist, we can't continue
+                // although in future we could interpolate to missing continuous
+                // values if we can work out what to do with all non-continuous ones,
+                // and if it's reasonable that frameAtIndexInterpolated make up new data points
+                if(!rowA || !rowB) {
+                    return null;
+                }
+
+                // for each column, try to interpolate values
+                return this.columns.reduce((map: ChartRow, {key, isContinuous}): ChartRow => {
+                    // only try to interpolate continuous non-primary columns
+                    const value: ChartScalar = isContinuous && key != primaryColumn
+                        ? ChartData.lerp(rowA.get(key), rowB.get(key), blend)
+                        : rowA.get(key);
+
+                    return map.set(key, value);
+                }, Map());
+            })
+            .filter(ii => ii != null);
+
+        return new ChartData(interpolatedFrame, this.columns);
+    }
+
+    /**
+     * Get the minimum non-null value in a column.
+     *
+     * @param {string|Array<string>|List<string>} columns
+     * The names of one or more columns to perform the operation on.
+     *
+     * @return {number|null} The minimum value, or null if no mininum value could be determined.
+     *
+     * @name min
+     * @kind function
+     * @inner
+     * @memberof ChartData
+     */
+
+    min(columns: ChartColumnArg): ?ChartScalar {
```

Comment:
I'd prefer to keep them separate in the API so the usage for the user is nicer. Also makes it more documentable to have `chartData.min` and `chartData.max` rather than `chartData.math` (can accept min which does x and max which does y). But I totally agree that the guts of those functions are duplicative, definitely a side-effect of sublimes multi-cursor editing. I might make a private `_aggregation` method that is called by each of those public aggregation methods, to remove the duplication, sound alright?